module/User/src/Controller/ApiAuthenticationController.php

```diff
@@ -23,31 +29,74 @@ class ApiAuthenticationController extends AbstractActionController
      */
     protected ApiAppService $apiAppService;
 
+    protected ApiAppAuthenticationMapper $apiAppAuthenticationMapper;
+
+    protected ApiAppMapper $apiAppMapper;
+
     /**
      * ApiAuthenticationController constructor.
      *
      * @param AclService $aclService
      * @param ApiAppService $apiAppService
+     * @param ApiAppMapper $apiAppMapper
      */
     public function __construct(
         AclService $aclService,
         ApiAppService $apiAppService,
+        ApiAppAuthenticationMapper $apiAppAuthenticationMapper,
+        ApiAppMapper $apiAppMapper,
     ) {
         $this->aclService = $aclService;
         $this->apiAppService = $apiAppService;
+        $this->apiAppAuthenticationMapper = $apiAppAuthenticationMapper;
+        $this->apiAppMapper = $apiAppMapper;
     }
 
-    public function tokenAction(): Response
+    public function tokenAction(): Response|ViewModel
     {
         $appId = $this->params()->fromRoute('appId');
+        $app = $this->apiAppMapper->findByAppId($appId);
+
+        if (null === $app) {
+            return $this->notFoundAction();
```

Comment:
It's good practice to first retrieve the identity such that some sort of ACL is performed before revealing information about our database contents.

Reply:
I'll change it but I don't think it is super common to have in external authorization applications. For example, Apple, Facebook, and Twitter will show an error page if the token (app id in our case) is not valid without having to sign in.